tests/test_CI.py

```diff
@@ -48,7 +48,7 @@ def dist_func():
         print(f"worker rank: {rank}")
 
     try:
-        shutil.rmtree("./test_logs")
+        shutil.rmtree("./test_logs", ignore_errors=True)
     except FileNotFoundError:
         pass
 
```

Comment:
Let's just create a temporary directory (with `tempfile`) and then not clean it up (will be discarded in CI anyway).

Reply:
Yeah I have a newer version of these tests on another branch that I think uses a tmp directory.